web/libs/datamanager/src/utils/urlJSON.js

```diff
@@ -0,0 +1,20 @@
+// Library that serializes and deserializes JSON data to/from URL-safe strings
+// Uses gzip to minimize the size of the serialized data
+
+import { gzip, ungzip } from 'pako'
+import { fromUint8Array, toUint8Array } from 'js-base64'
+import { packJSON } from './packJSON';
+
+export const deserializeJsonFromUrl = (s) => {
+    // Fall back to packJSON if deserialization fails; this probably means
+    // we are deserializing an old virtual view. TODO(jo): deprecate packJSON entirely
+    try {
+        return JSON.parse(ungzip(toUint8Array(s), { to: 'string' }));
+    } catch (e) {
+        console.log('Error deserializing gzipped data:', e);
+        console.log('falling back to packJSON.');
+        return packJSON.parse(s)
+    }
+}
+
+export const serializeJsonForUrl = (s) => fromUint8Array(gzip(JSON.stringify(s)), true)
```

Comment:
note that this is url-safe: https://github.com/dankogai/js-base64/blob/main/base64.js#L102-L110